sharding-and-replication.md

```diff
@@ -0,0 +1,156 @@
+# Sharding and Replication
+
+This is a guide how to setup sharding and replication for Jaeger data.
+This guide uses [clickhouse-operator](https://github.com/Altinity/clickhouse-operator) to deploy
+the storage.
+
+## Sharding
+
+Sharding is a feature that allows splitting the data into multiple Clickhouse nodes to
+increase throughput and decrease latency.
+The sharding feature uses `Distributed` engine that is backed by local tables.
+The distributed engine is a "virtual" table that does not store any data. It is used as 
+an interface to insert and query data.
+
+To setup sharding run the following statements on all nodes in the `sharded` cluster.
+The "local" tables have to be created on the nodes before the distributed table.
+
+```sql
+CREATE TABLE jaeger_spans_global AS jaeger_spans ENGINE = Distributed(sharded, default, jaeger_spans, rand());
```

Comment:
It would be good to use `traceID` as a sharding key, but it is `string` and that cannot be used.

Reply:
In that case, you can use `murmurHash3_64(traceID)` as sharding key.

Reply:
based on your experience with CH, is it good to use sharding by traceID?

Reply:
We use `rand()`as the sharding key in production but `cityHash64` is pretty fast too. We can do some quick testing to see how they performs.

Reply:
I have booked a ticket for it https://github.com/pavolloffay/jaeger-clickhouse/issues/35